Fluid/Ast/IncludeStatement.cs

```diff
@@ -31,15 +34,37 @@ public IncludeStatement(Expression path)
                 throw new FileNotFoundException(relativePath);
             }
 
-            string partialContent;
+            if (!context.AmbientValues.ContainsKey(FluidParserFactoryKey))
+            {
+                throw new ArgumentException($"The '{FluidParserFactoryKey}' key was not present in the AmbientValues dictionary.", nameof(context));
+            }
+            if (!context.AmbientValues.ContainsKey(FluidTemplateKey))
+            {
+                throw new ArgumentException($"The '{FluidTemplateKey}' key was not present in the AmbientValues dictionary.", nameof(context));
+            }
+
+            FluidParserFactory factory = (FluidParserFactory)context.AmbientValues[FluidParserFactoryKey];
+            IFluidParser parser = factory.CreateParser();
+            IFluidTemplate template = (IFluidTemplate)context.AmbientValues[FluidTemplateKey];
```

Comment:
Can't it be created from the factory? Even if we need to add something to the factory?

Reply:
The template isn't available from the parser factory.

Reply:
Then just put the type in the `context.AmbientValues[FluidTemplateKey]` and call Activator.CreateInstance when you need the template instance. The goal is to prevent the template allocation everytime a filter is used.

Reply:
And also because if you call include multiple times you want a new template instance every time.

Reply:
My latest commit uses parser and template factories from ambient values if they are present, otherwise it uses what seems like sane defaults. So nothing is "globally" configurable per-se. I set the ambient values in the MVC renderer, but do not set them or need to in the tests.